tls-client/main.cpp

```diff
@@ -132,6 +133,8 @@ class HelloHTTPS {
         mbedtls_x509_crt_free(&_cacert);
         mbedtls_ssl_free(&_ssl);
         mbedtls_ssl_config_free(&_ssl_conf);
+        _tcpsocket->close();
```

Comment:
In the latest version the constructor remained the same and the function closes the connection before exiting and this approach resolves my above concern as well.